File: platforms/python/examples/core/texture_mip_chain/prepare.py
from fragmentcolor import Renderer, TextureFormat, TextureMipChain

# Raw pixel path -- positional args: prepare(bytes, format, size).
raw_rgba = [200] * (8 * 8 * 4)
chain = TextureMipChain.prepare(raw_rgba, TextureFormat.Rgba8UnormSrgb, [8, 8])

# Encoded path (no size) -- prepare decodes the image and infers dimensions.
chain_raw = TextureMipChain.prepare(raw_rgba, TextureFormat.Rgba8UnormSrgb, [8, 8])

# Hand the chain to the unified create_texture entry -- same vocabulary.
renderer = Renderer()
texture = renderer.create_texture(chain)

File: platforms/web/examples/core/texture_mip_chain/prepare.js
import { Renderer, TextureFormat, TextureMipChain } from "fragmentcolor";

// Encoded path — single tuple, no extra method.
const chain = TextureMipChain.prepare((encoded_png_bytes, TextureFormat.Rgba8UnormSrgb));

// Raw pixel path — same method, just include the size in the tuple.
const chain_raw = TextureMipChain.prepare((
    raw_rgba.asSlice(),
    TextureFormat.Rgba8UnormSrgb,
    [8, 8],
));

// Hand the chain to the unified create_texture entry — same vocabulary.
const renderer = new Renderer();
const texture = await renderer.createTexture(chain);